### ISSUE TITLE
TypeORM returns incorrect values for time type columns when database is SQL Server

### ISSUE THREAD

**Issue by penttin5** [typeorm/typeorm#10403]
### Issue description

TypeORM returns incorrect values for time type columns when database is SQL Server

### Expected Behavior

When reading a time column from db with repository.find or dataSource.query, both should return the value stored in the db.

### Actual Behavior

If TypeORM's useUTC is true, and local timezone is UTC+3 (utc+2 + daylight savings time = utc+3):
* reading an entity with time column from db with repository.find returns incorrect time. E.g. if db contains '08:00:00', the entity returned from repository.find contains value '10:00:00'
* reading a time column from the db with dataSource.query returns Java Date `1970-01-01T08:00:00.000Z`, which has the correct time, but I'd expect it to return a time string the same way as it does with entities.

If useUTC is false:
* reading an entity with time column from db with repository.find returns correct time. E.g. if db contains '08:00:00', the entity returned from repository.find contains value '08:00:00'
* reading a time column from the db with dataSource.query returns incorrect Java Date `1970-01-01T06:00:00.000Z`, I'd also expect it to return a time string the same way as it does with entities.

### Steps to reproduce

1. set useUTC=true or false in TypeORM's options
2. define a time column to entity
   ```ts
   @Entity()
   export class TimeTestEntity {
     @Column({ type: 'time' })
     timeString!: string
   ```
3. save time value into the db 
   ```ts
      await dataSource.query("insert into test_entity (timeString) values ('08:00:00')")
   ```
4. read the value from the db with repository.find and with raw query
   ```ts
      const entitiesInDb = await dataSource.getRepository(TimeTestEntity).find()
      const rawQueryResults = await dataSource.query('select * from time_test_entity')
   ```
if useUTC is true:
  * entitiesInDb contains wrong timeString: `entitiesInDb[0].timeString === '10:00:00'`
  * the raw query results contains correct time `new Date('1970-01-01T08:00:00.000Z')`, but the type is Javascript Date instead of string

if useUTC is false:
  * entitiesInDb contains correct timeString: `entitiesInDb[0].timeString === '08:00:00'`
  * the raw query results contains incorrect time `new Date('1970-01-01T06:00:00.000Z')`, and the type is Javascript Date instead of string

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  macOs        |
| Node.js version     | 16.14.0  |
| Typescript version  | 5.2.2 |
| TypeORM version     | 0.3.17 |
| mssql | 9.3.0 |

### Additional Context

Where conditions in repository.find seem to work correctly for both useUTC=true and false.
E.g if database contains time 08:00:00
```ts
      const entity = await dataSource.getRepository(TimeTestEntity).findOneByOrFail({ timeString: '08:00:00' })
```
returns the correct entity, even though timeString value on the returned entity is wrong

We could circumvent the bug by adjusting the time values in ValueTransformer, but https://github.com/typeorm/typeorm/issues/10397 prevents this.

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [ ] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [X] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by qayyum22** [comment]
Hey @penttin5, i'm new to open source , i would like to work on its, could you please assign it to me?

**Comment by alumni** [comment]
The main issue with dates is that each client treats them differently, often incorrectly.

I tend to set the environment variable TZ=UTC so that Node.JS works with UTC and I keep the DB's timezone also as UTC. This simplifies things.

For SqlServer, `useUTC` is passed to the `mssql` client.

For Postgres, `process.env.PGTZ` is set to `UTC` in: https://github.com/typeorm/typeorm/blob/5003aaa7c5143a4a718f1d9c3de227cc12e19dff/src/driver/postgres/PostgresDriver.ts#L313-L315

Feel free to submit a PR if you know how to instruct the clients to use the correct timezone.

**Comment by tolgap** [comment]
Yeah we're not setting `process.env.TZ = 'UTC'` anywhere in our application code. What ends up happening is `pg-types` and its dependency `postgres-date` ends up using `JSDate` to construct the `my_date` column. Since that column does not include any timezone offset/info, the JSDate uses the machine timezone (obviously).

The workaround could indeed be to just always ensure that the application is set to `TZ = 'UTC'` and typeorm is set to `useUTC: true`.

I appreciate your help on my end @alumni ❤

**Comment by alumni** [comment]
See also #11498
